app/javascript/packages/phone-input/index.ts

```diff
@@ -202,14 +196,13 @@ export class PhoneInputElement extends HTMLElement {
       return;
     }
 
-    const isInvalidCountry = !isValidNumberForRegion(phoneNumber, countryCode);
```

Comment:
Per previous comment, I wouldn't expect the validation logic to change in this pull request, only the message that's being shown. Maybe we can extract a method to call when the number is invalid, which will choose the correct messages based on the selected country code?

```ts
const isInvalidCountry = !isValidNumberForRegion(phoneNumber, countryCode);
if (isInvalidCountry) {
  textInput.setCustomValidity(this.getInvalidFormatMessage(countryCode));
}

const isInvalidPhoneNumber = !isPhoneValid(phoneNumber, countryCode);
if (isInvalidPhoneNumber) {
  textInput.setCustomValidity(this.getInvalidFormatMessage(countryCode));
}
```

```ts
getInvalidFormatMessage(countryCode: CountryCode): string {
  return countryCode === 'US'
    ? this.strings.invalid_phone_us || ''
    : this.strings.invalid_phone_international || '';
}
```